Application/Business/AttributeGroup/AttributeGroupServices.cs

```diff
@@ -22,34 +27,40 @@ public async Task<int> Count()
     {
         return await _attributeRepo.CountAsync();
     }
-    public async Task Create(AttributeGroupEntity attributeGroup)
+    public async Task Create(AttributeGroupEntity attributeGroup){
+        await _attributeRepo.CreateAsync(attributeGroup);
+    }
+    public async Task Update(Guid attributeGroupid, AttributeGroupEntity attributeGroup)
     {
-        var attList = List();
-        var check = false;
-        foreach (var attgrp in attList.Result)
+        var target = await GetById(attributeGroupid);
+        if(target is not null)
         {
-            if (attgrp.Name == attributeGroup.Name)
+            Type entityType = typeof(AttributeGroupEntity);
+            foreach (PropertyInfo propertyInfo in entityType.GetProperties())
             {
-                check = true;
-                break;
+                // Get the value of the property in attributeGroup
+                object value = propertyInfo.GetValue(attributeGroup);
+
+                // Check if the value is not null and update the corresponding property in target
+                if (value != null && value.ToString() != "00000000-0000-0000-0000-000000000000")
+                {
+                    propertyInfo.SetValue(target, value);
+                }
             }
+            //Insert changes here
+            await _attributeRepo.UpdateAsync(target);
         }
-        if (!check)
+        else
         {
-            var attGrp = new AttributeGroupEntity
-            {
-                Name = attributeGroup.Name,
-                Effect = attributeGroup.Effect
-            };
-            await _attributeRepo.CreateAsync(attGrp);
+            throw new NotFoundException("Attribute group not exist");
```

Comment:
let check if Attribute group not exist ==> return, then do logic to reduce nested block